src/Polly.Shared/Timeout/TimeoutEngine.cs

```diff
@@ -47,8 +47,14 @@ internal static partial class TimeoutEngine
                         .Run(() =>
                             action(context, combinedToken)       // cancellation token here allows the user delegate to react to cancellation: possibly clear up; then throw an OperationCanceledException.
                             , combinedToken);           // cancellation token here only allows Task.Run() to not begin the passed delegate at all, if cancellation occurs prior to invoking the delegate.  
-
-                        actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        try
+                        {
+                            actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        }
+                        catch (AggregateException ex) when (ex.InnerExceptions.Count == 1)
```

Comment:
When an exception is thrown inside the task, that exception is wraped in an AggregateException when we use Task.Wait().

In this scenario the AggregateException will always have one inner exception. The inner exception itself can always be an AggregateException with multiple inner exceptions.

This will happen if delegate itself throws an AggregateException with many inner exceptions and I dont want to mess with that in this change. The intention is only to preserve the exception behaviour of the delegate and remove the inconsistency with running it inside a task.

This is also the reason that I am not use .Flatten(), as that would change the exception thrown by the delegate.